### PR TITLE
Task #173069 feat: Update infra extensions code used to make Joomla 4 compatible

### PR DESCRIPTION
**Pull request by ankush-maherwal**

### CODE REVIEW

File: src/techjoomla/tjnotifications/tjnotifications.php [hunk 1]
```diff
@@ -1,3 +1,6 @@
+use Joomla\CMS\MVC\Model\BaseDatabaseModel;
```
Comment (manojLondhe) [inline src/techjoomla/tjnotifications/tjnotifications.php:1]: this needs to be after opening php tags

Comment (manojLondhe) [reply src/techjoomla/tjnotifications/tjnotifications.php:1]: probably bcoz this file does not have

defined('_JEXEC') or die();

File: src/com_tjnotifications/site/views/preferences/tmpl/default.php [hunk 2]
```diff
@@ -16,7 +17,7 @@
 
 <script src="/jquery.min.js"></script>
 <script type="text/javascript">
-	const tjnBaseurl = "<?php echo JUri::root();?>";
```
Comment (manojLondhe) [inline src/com_tjnotifications/site/views/preferences/tmpl/default.php:19]: load inline js using $doc->addScriptDeclaration()

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/plugins/privacy/tjnotification/tjnotification.php [hunk 3]
```diff
@@ -10,6 +10,9 @@
 
 // No direct access.
 defined('_JEXEC') or die();
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\User\User;
+use Joomla\CMS\Table\User;
```
[the remark this answers is not in the document]
Comment (ankush-maherwal) [reply src/plugins/privacy/tjnotification/tjnotification.php]: Done

File: src/techjoomla/tjnotifications/tjnotifications.php (hunk 1 above)
Comment (ankush-maherwal) [reply src/techjoomla/tjnotifications/tjnotifications.php]: done

File: src/com_tjnotifications/site/views/preferences/tmpl/default.php (hunk 2 above)
Comment (ankush-maherwal) [reply src/com_tjnotifications/site/views/preferences/tmpl/default.php:19]: done

File: src/com_tjnotifications/admin/sql/install.mysql.utf8.sql [hunk 4]
```diff
@@ -26,9 +26,9 @@ CREATE TABLE IF NOT EXISTS `#__tj_notification_template_configs` (
   `template_id` int(11) NOT NULL DEFAULT 0,
   `backend` varchar(50) NOT NULL DEFAULT '',
   `language` char(7) NOT NULL DEFAULT '*',
-  `subject` text NOT NULL DEFAULT '',
-  `body` text NOT NULL DEFAULT '',
-  `params` text NOT NULL DEFAULT '',
+  `subject` text NOT NULL,
+  `body` text NOT NULL,
+  `params` text NOT NULL,
```
Comment (manojLondhe) [inline src/com_tjnotifications/admin/sql/install.mysql.utf8.sql:31]: params could be null

`params` text NOT NULL DEFAULT '' =>`params` text  DEFAULT NULL

makes more sense

**Pull request merged**